tools/benchmarksAnalyzer/src/main/kotlin/org/jetbrains/analyzer/SummaryBenchmarksReport.kt

```diff
@@ -0,0 +1,423 @@
+                    percentsList = percentsList.map{it + precision}
+                    geomeanValue = Statistics.geometricMean(percentsList) - precision
+                }
+
+                append("$bucketName: Maximum = ${formatValue(maxValue, true)}," +
+                        " Geometric mean = ${formatValue(geomeanValue, true)}")
+            }
+        }
+
+        override fun renderStatusSummary() {
+            append("Status summary")
+            append(headerSeparator)
+            if (failedBenchs.isEmpty()) {
+                append("All benchmarks passed!")
+            }
+            if (!failedBenchs.isEmpty() || !addedBenchs.isEmpty() || !removedBenchs.isEmpty()) {
+                printBucketInfo(failedBenchs, "Failed benchmarks")
+                printBucketInfo(addedBenchs, "Added benchmarks")
+                printBucketInfo(removedBenchs, "Removed benchmarks")
+            }
+            append("Total becnhmarks number: ${mergedReport.keys.size}")
+            append()
+
+            if (!newFailures.isEmpty() || !newPasses.isEmpty()) {
+                append("Changes in status")
+                append(headerSeparator)
+                printStatusChangeInfo(newFailures, "New failures")
+                printStatusChangeInfo(newPasses, "New passes")
+                append()
+            }
+        }
+
+        override fun renderPerformanceSummary() {
+            if (!regressions.isEmpty() || !improvements.isEmpty()) {
+                append("Performance summary")
+                append(headerSeparator)
+                printPerformanceBucket(regressions, "Regressions")
+                printPerformanceBucket(improvements, "Improvements")
+                append()
+            }
+        }
+
+        private fun formatValue(number: Double, isPercent: Boolean = false): String {
+            if (isPercent) {
+                return format(number, 2) + "%"
+            }
+            return format(number)
+        }
+
+        private fun formatColumn(content:String, isWide: Boolean = false): String {
+            val width = if (isWide) 50 else 25
+            return content.padEnd(width, ' ')
+        }
+
+        private fun printBenchmarksDetails(fullSet: MutableMap<String, SummaryBench>,
+                                           bucket: Map<String, ScoreChange>? = null) {
+            bucket?.let {
+                // There are changes in performance.
+                // Output changed benchmarks.
+                for ((name, change) in bucket) {
+                    append("${formatColumn(name, true)} " +
+                            "${formatColumn(fullSet[name]!!.first!!.toString())} " +
+                            "${formatColumn(fullSet[name]!!.second!!.toString())} " +
+                            "${formatColumn(change.first.toString() + " %")} " +
+                            "${formatColumn(change.second.toString())}")
+                    fullSet.remove(name)
```

Comment:
It is really unintuitive that printing function is also removing entries from its argument.